### PR TITLE
Python panel docs updates

### PR DESCRIPTION
**Pull request by Br2850**

### CODE REVIEW

File: docs/source/plugins/developing_plugins.rst [hunk 1]
```diff
@@ -2129,6 +2233,565 @@ plugin. Panels access these secrets via the `ctx.secrets` dict:
...
+            )
+
+
+    # Utility function to enhance styling of navigation buttons
+
+
+    def add_panel_navigation(
+        panel, left=True, right=False, on_left=None, on_right=None
+    ):
+        base_btn_styles = {
```
Comment (brimoor) [inline docs/source/plugins/developing_plugins.rst:2544]: @imanjra could you update this to show how to use the builtin next/prev buttons?

File: docs/source/plugins/developing_plugins.rst [hunk 2]
```diff
@@ -2129,6 +2233,565 @@ plugin. Panels access these secrets via the `ctx.secrets` dict:
...
+    def reset(self, ctx):
+        # clear any saved views on operator
+        ctx.ops.clear_view()
+        self.on_load(ctx)
+
+    def render(self, ctx):
+        panel = types.Object()
+
+        # Bar Chart - Histogram
+        panel.plot(
```
[the remark this answers is not in the document]
Comment (imanjra) [reply docs/source/plugins/developing_plugins.rst:2365]: I believe all the types should support data if it supports the state. Was there a case where this was not true for you?

[the remark this answers is not in the document]
Comment (Br2850) [reply docs/source/plugins/developing_plugins.rst:2365]: @imanjra @brimoor Here's the source code for that, the setting plot data part should be in the `on_load` function: https://github.com/voxel51/fiftyone-plugins/blob/8f71a2241d3b2390b949c5ecd9deba9d8da65a52/plugins/panel-examples/__init__.py#L555

You can just copy paste that code and it'll work

**Comment by imanjra** [comment]
@Br2850 you can use command below to downgrade some dependencies need to run docs build successfully:

```sh
pip install Sphinx==3.5.4 sphinx-copybutton==0.4.0 sphinx-rtd-theme==1.3.0 sphinx-tabs==1.2.1 sphinxcontrib-applehelp==1.0.4 sphinxcontrib-devhelp==1.0.2 sphinxcontrib-htmlhelp==2.0.1 sphinxcontrib-jquery==4.1 sphinxcontrib-jsmath==1.0.1 sphinxcontrib-napoleon==0.7 sphinxcontrib-qthelp==1.0.3 sphinxcontrib-serializinghtml==1.1.5
```

**Comment by Br2850** [comment]
@imanjra Okay dope, noted for next time. It built fine with `5.0.0` but will use that command in the future to run a suite of dependencies that work together

File: docs/source/plugins/developing_plugins.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (brimoor) [reply docs/source/plugins/developing_plugins.rst:2414]: Update: I'm now *unconfused* about state vs data. They're deep merged into the same structure when rendering

**Review by brimoor — APPROVED** [review]
LGTM

**Pull request merged**